scripts/DialogData.py

```diff
@@ -14,44 +14,193 @@
-
-X_PLACEHOLDER = '<x>'
+from EntityData import EntityData
+from NodeData import NodeData
+import unicodedata, unidecode
+from wawCommons import printf, eprintf, toIntentName
+X_PLACEHOLDER = u'&lt;x&gt;'
 
 class DialogData(object):
-    """ Represents complete Dialog data structure. """
+    """ DialogData represents complete dialog (intents, entities, nodes, labels).
+    The data are collected across all processed T2C source.
+        _domains keeps track how domains are affiliated to nodes as we generate separate xls for each domain
+        concept of domain correspods to a single Excel Workbook
+    """
 
+    def __init__(self, config):
+        # we store on bag of intents and one bag of entities (each results in one file all in corresponding directory
+        self._labelsMap = {} # key: lable, value - node name - translation table for generating jumps
 
-    def __init__(self):
-        self._intents = {}  # key: intent name, value: IntentData object
-        self._entities = {}  # key: entity name, value: list of all Dialog entity options
-        self._domains = {}  # key: domain name, value: list of all associated Dialog intents
+        self._entities = {}  # key: entity name, value: list of all Dialog entity options; entities is a flat list over all the domains
+        self._intents = {}   # key: intent name, value: IntentData object; intents is a flat list over all the domains
 
+        # nodes are grouped based on domain
+        self._nodes = {}     # key: node name, value: list of all associated dialog nodes
+        self._domains = {}   # key: domain name, value: list of all associated nodes with the given domain
 
-    def getEntity(self, entityName):
-        if entityName not in self._entities:
-            self._entities[entityName] = []
-        return self._entities[entityName]
+        self._config= config            #we need config to get NAME_POLICY, verbosity,..
+        self._VERBOSE = hasattr(config, 'common_verbose')
+        self._NAME_POLICY = 'soft'      # TBD: enable to set the NamePolicy from config file
 
+    #  LABEL
+    #******************************************
 
-    def getAllEntities(self):
-        return self._entities
+    def addLabel(self, label, node_name):
+        self._labelsMap[label] = node_name
+
+    def isLabel(self, label):
+        # Checks if label exists
+        return (label in self._labelsMap)
+
+    def getLabelsMap(self):
+        """ Return map with GoTo labels as keys and target  node_names as values. """
+        return self._labelsMap
 
+    #  ENTITY
+    #******************************************
 
-    def getIntentData(self, intentName, domainName=None):
-        if domainName is not None:
-            if domainName not in self._domains:
-                self._domains[domainName] = []
-            if intentName not in self._domains[domainName]:
-                self._domains[domainName].append(intentName)
+    def createEntity(self, entity_name):
```

Comment:
Why this function behaves differently than the `createIntent` one? (Returns None)

Reply:
Agree, code was unified and checks added for situations where attempts to create nodes, intents or entities of the same name are made